### PR TITLE
Convert micrometer Counters to FunctionCounters

### LINKED ISSUE #4540: Assess which micrometer metrics `Counter`s can be converted to `FunctionCounter`s
> If these are monotonically increasing counters, then it would be better to use a [FunctionCounter](https://docs.micrometer.io/micrometer/reference/concepts/counters.html#_function_tracking_counters) instead of a Gauge.  Not sure if they are, need to investigate.  The micrometer code to instrument caches uses FunctionCounter internally

_Originally posted by @keith-turner in https://github.com/apache/accumulo/pull/4518#discussion_r1591458201_
            
            
We should look at the places we are using a `Counter` and determine if it is a good idea to convert them to `FunctionCounter`.

### PR DESCRIPTION
**Pull request by DomGarguilo**
Fixes #4540

This PR converts the usage of `Counter` to `FunctionCounter` where it makes sense.

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerUpdateMetrics.java [hunk 1]
```diff
@@ -68,10 +69,14 @@ public void addMutationArraySize(long value) {
 
   @Override
   public void registerMetrics(MeterRegistry registry) {
-    permissionErrorsCounter = registry.counter(METRICS_UPDATE_ERRORS, "type", "permission");
-    unknownTabletErrorsCounter = registry.counter(METRICS_UPDATE_ERRORS, "type", "unknown.tablet");
-    constraintViolationsCounter =
-        registry.counter(METRICS_UPDATE_ERRORS, "type", "constraint.violation");
+    FunctionCounter.builder("tabletServerUpdateErrors", permissionErrorsCount, AtomicLong::get)
```
Comment (dlmarion) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerUpdateMetrics.java:72]: I think the names still need to be in the standard form, `accumulo.tserver.updates.errors.permission` for example. Do these need to be unique, is that why you are not using `METRICS_UPDATE_ERRORS` for these?

**Review by dlmarion — APPROVED** [review]

File: server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerUpdateMetrics.java (hunk 1 above)
Comment (DomGarguilo) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/metrics/TabletServerUpdateMetrics.java]: I changed them back to use `METRICS_UPDATE_ERRORS` for the name like before. I changed these my mistake.

**Comment by cshannon** [comment]
I just checked the CI build and noticed that this commit is causing a new test failure in `MetricsIT` in the `confirmMetricsPublished` test so something probably needs tweaking:

```
org.opentest4j.AssertionFailedError: Found accumulo metric not in expectedMetricNames or flakyMetricNames: accumulo.tserver.compactions.majc.paused
	at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:38)
	at org.junit.jupiter.api.Assertions.fail(Assertions.java:138)
	at org.apache.accumulo.test.metrics.MetricsIT.lambda$confirmMetricsPublished$1(MetricsIT.java:133)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.apache.accumulo.test.metrics.MetricsIT.confirmMetricsPublished(MetricsIT.java:124)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

**Comment by cshannon** [comment]
Looks like @EdColeman recently updated/added that test so he might have some insight into it if not sure.

**Comment by EdColeman** [comment]
Addressed in #4576

**Pull request merged**